test/test_complex.py

```diff
@@ -21,6 +35,111 @@ def test_dtype_inference(self, device, dtype):
         torch.set_default_dtype(default_dtype)
         self.assertEqual(x.dtype, torch.cdouble if dtype == torch.float64 else torch.cfloat)
 
+    @dtypes(torch.float32, torch.float64)
+    def test_torch_complex(self, device, dtype):
+        real = torch.tensor([1, 2], device=device, dtype=dtype)
+        imag = torch.tensor([3, 4], device=device, dtype=dtype)
+        z = torch.complex(real, imag)
+        complex_dtype = torch.complex64 if dtype == torch.float32 else torch.complex128
+        self.assertEqual(torch.tensor([1.0 + 3.0j, 2.0 + 4.0j], dtype=complex_dtype), z)
+
+    @dtypes(torch.float32, torch.float64)
+    def test_torch_complex_polar(self, device, dtype):
+        abs = torch.tensor([1, 2], device=device, dtype=dtype)
+        angle = torch.tensor([np.pi / 2, 5 * np.pi / 4], device=device, dtype=dtype)
```

Comment:
Although instead of using numpy here I think you can just use math.pi?

Reply:
Better add a few more angles, too: 0, pi/4, pi/2, pi, -pi/4, -pi/2, -pi.

Reply:
Added a few more but slightly different examples.